File: src/org/telegram/messenger/GroqAiService.ts
/**
 * Telegram Official GroqAiService
 * Connects to LLM endpoints (Groq / Gemini / Server proxy) for Gulf & Arabic dialect generation
 */

export interface AiGenerationResult {
  success: boolean;
  reply?: string;
  error?: string;
}

export class GroqAiService {
  private static instance: GroqAiService;
  private apiKey: string = '';
  private defaultPersona: string = 'أنت مساعد ذكي ولطيف، رد دائماً بلهجة خليجية بيضاء مختصرة، ودودة، ومفيدة جداً.';

  public static getInstance(): GroqAiService {
    if (!GroqAiService.instance) {
      GroqAiService.instance = new GroqAiService();
    }
    return GroqAiService.instance;
  }

  public setApiKey(key: string): void {
    this.apiKey = key;
  }

  /**
   * Generates Gulf dialect reply using AI
   */
  public async generateGulfResponse(userMessage: string, customPersona?: string): Promise<AiGenerationResult> {
    try {
      const prompt = customPersona || this.defaultPersona;

      // Try server-side AI proxy first
      const res = await fetch('/api/ai/chat', {
        method: 'POST',
        headers: { 'Content-Type': 'application/json' },
        body: JSON.stringify({
          message: userMessage,
          persona: prompt,
        }),
      });

      if (res.ok) {
        const data = await res.json();
        if (data.reply || data.text) {
          return { success: true, reply: data.reply || data.text };
        }
      }

      // Fallback: direct Groq API if client key provided
      if (this.apiKey) {
        const groqRes = await fetch('https://api.groq.com/openai/v1/chat/completions', {
          method: 'POST',
          headers: {
            'Content-Type': 'application/json',
            Authorization: `Bearer ${this.apiKey}`,
          },
          body: JSON.stringify({
            model: 'llama3-8b-8192',
            messages: [
              { role: 'system', content: prompt },
              { role: 'user', content: userMessage },
            ],
            temperature: 0.7,
          }),
        });

        if (groqRes.ok) {
          const groqData = await groqRes.json();
          const reply = groqData.choices?.[0]?.message?.content;
          if (reply) return { success: true, reply };
        }
      }

      // Smart heuristic fallback
      return {
        success: true,
        reply: `أهلاً بك يا غالي! تسلم على رسالتك: "${userMessage.slice(0, 30)}..."، أبشر بكل خير ويسعدني خدمتك دائماً! 🌟`,
      };
    } catch (e: any) {
      return {
        success: false,
        error: e.message || 'تعذر الحصول على رد الذكاء الاصطناعي',
      };
    }
  }
}

File: TMessagesProject/src/main/java/org/telegram/messenger/GroqAiService.java
package org.telegram.messenger;

import org.json.JSONArray;
import org.json.JSONObject;
import java.io.IOException;
import okhttp3.Call;
import okhttp3.Callback;
import okhttp3.MediaType;
import okhttp3.OkHttpClient;
import okhttp3.Request;
import okhttp3.RequestBody;
import okhttp3.Response;

/**
 * GroqAiService - Integration with Groq Cloud LLM (llama3-8b-8192) for Gulf/Arabic Smart Assistant
 */
public class GroqAiService {

    private static final String GROQ_ENDPOINT = "https://api.groq.com/openai/v1/chat/completions";
    private static final OkHttpClient client = new OkHttpClient();

    public interface GroqCallback {
        void onSuccess(String reply);
        void onError(String error);
    }

    public static void generateGulfReply(String apiKey, String userMessage, String personaPrompt, GroqCallback callback) {
        try {
            JSONObject json = new JSONObject();
            json.put("model", "llama3-8b-8192");
            json.put("temperature", 0.7);

            JSONArray messages = new JSONArray();
            
            JSONObject systemMsg = new JSONObject();
            systemMsg.put("role", "system");
            systemMsg.put("content", personaPrompt != null && !personaPrompt.isEmpty() ? personaPrompt : "أنت مساعد ذكي ولطيف، رد دائماً بلهجة خليجية بيضاء مختصرة، ودودة، ومفيدة.");
            messages.put(systemMsg);

            JSONObject userMsg = new JSONObject();
            userMsg.put("role", "user");
            userMsg.put("content", userMessage);
            messages.put(userMsg);

            json.put("messages", messages);

            RequestBody body = RequestBody.create(json.toString(), MediaType.parse("application/json; charset=utf-8"));
            Request request = new Request.Builder()
                    .url(GROQ_ENDPOINT)
                    .header("Authorization", "Bearer " + apiKey)
                    .header("Content-Type", "application/json")
                    .post(body)
                    .build();

            client.newCall(request).enqueue(new Callback() {
                @Override
                public void onFailure(Call call, IOException e) {
                    if (callback != null) callback.onError(e.getMessage());
                }

                @Override
                public void onResponse(Call call, Response response) throws IOException {
                    if (!response.isSuccessful()) {
                        if (callback != null) callback.onError("HTTP " + response.code() + ": " + response.message());
                        return;
                    }
                    try {
                        String bodyStr = response.body().string();
                        JSONObject resObj = new JSONObject(bodyStr);
                        String reply = resObj.getJSONArray("choices")
                                .getJSONObject(0)
                                .getJSONObject("message")
                                .getString("content");
                        if (callback != null) callback.onSuccess(reply);
                    } catch (Exception e) {
                        if (callback != null) callback.onError("Failed to parse JSON: " + e.getMessage());
                    }
                }
            });
        } catch (Exception e) {
            if (callback != null) callback.onError(e.getMessage());
        }
    }
}
